File: AndroidApp/PayIn/app/src/main/java/com/mobile/application/payin/dto/arguments/MobileConfigurationArguments.java
package com.mobile.application.payin.dto.arguments;

import android.content.Context;
import android.net.wifi.WifiManager;
import android.os.Build;
import android.telephony.TelephonyManager;

public class MobileConfigurationArguments {
    public String DeviceManufacturer = "";
    public String DeviceModel        = "";
    public String DeviceName         = "";
    public String DeviceSerial       = "";
    public String DeviceId           = "";
    public String DeviceOperator     = "";
    public String DeviceImei         = "";
    public String DeviceMac          = "";
    public String OperatorSim        = "";
    public String OperatorMobile     = "";

    public MobileConfigurationArguments(Context context) {
        TelephonyManager tm = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);

        String mac = ((WifiManager) context.getSystemService(Context.WIFI_SERVICE)).getConnectionInfo().getMacAddress();

        DeviceManufacturer = Build.MANUFACTURER;
        DeviceModel        = Build.MODEL;
        DeviceName         = Build.DISPLAY;
        DeviceSerial       = Build.SERIAL;
        DeviceId           = Build.ID;
        DeviceOperator     = tm.getNetworkOperatorName();
        try {
            DeviceImei = tm.getDeviceId();
        } catch (SecurityException e) { DeviceImei = ""; }
        DeviceMac          = mac == null ? "" : mac;
        OperatorSim        = tm.getSimOperatorName();
        OperatorMobile     = tm.getNetworkOperatorName();
    }
}

File: BusinessLogic/Xp.Application/Arguments/MobileConfigurationArguments.cs
using System.ComponentModel.DataAnnotations;
using Xp.Common.Dto.Arguments;

namespace Xp.Application.Arguments
{
  public abstract class MobileConfigurationArguments : IArgumentsBase
  {
		public string DeviceManufacturer { get; set; }
		public string DeviceModel        { get; set; }
		public string DeviceName         { get; set; } 
		public string DeviceSerial       { get; set; }
		public string DeviceId           { get; set; }
		public string DeviceOperator     { get; set; }
		public string DeviceImei         { get; set; }
		public string DeviceMac          { get; set; }
		public string OperatorSim        { get; set; }
		public string OperatorMobile     { get; set; }

		#region Constructors
		public MobileConfigurationArguments(string deviceManufacturer, string deviceModel, string deviceName, string deviceSerial, string deviceId, string deviceOperator, string deviceImei, string deviceMac, string operatorSim, string operatorMobile)
		{
			DeviceManufacturer = deviceManufacturer ?? "";
			DeviceModel = deviceModel ?? "";
			DeviceName = deviceName ?? "";
			DeviceSerial = deviceSerial ?? "";
			DeviceId = deviceId ?? "";
			DeviceOperator = deviceOperator ?? "";
			DeviceImei = deviceImei ?? "";
			DeviceMac = deviceMac ?? "";
			OperatorSim = operatorSim ?? "";
			OperatorMobile = operatorMobile ?? "";
		}
		#endregion Constructors
  }
}
